### PR TITLE
Group eslint dependencies in dependabot

### PR DESCRIPTION
**Pull request by slifty**
This PR adds an eslint grouping in dependabot so eslint related packages will get updated at the same time.

### CODE REVIEW

**Comment by slifty** [comment]
I opened it because the eslint-typescript library recently updated and requires eslint 8, which means the PR breaks (#611).
 
I thought "oh clearly there must be an eslint 8 PR somewhere else, and we need these grouped for the PRs to not break" but then realized that no, it's locked in with react app!

Which is to say, this PR will be more useful after implementing #498.  It isn't causing any harm until that point, however, and so I'd suggest we should merge it regardless.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -24,6 +24,9 @@ updates:
           - 'react-dom'
           - '@types/react'
           - '@types/react-dom'
+      eslint:
+        patterns:
+          - '*eslint*'
```
Comment (reefdog) [inline .github/dependabot.yml:29]: This worries me as potentially too broad. Can we just make it `eslint*` with the same practical results and less risk of sweeping up something unrelated?

Comment (slifty) [reply .github/dependabot.yml:29]: I copied the decisions made in the `service` repo and don't think it makes sense to have different rules for the same intention.

See the conversation around this change from there: https://github.com/PhilanthropyDataCommons/service/pull/592#discussion_r1393432857

What do you think?

Comment (reefdog) [reply .github/dependabot.yml:29]: Perfect! I was going to ask if it was too much trouble to do precisely that check. (I do still think `eslint*` is better than `*eslint*`, but we're arguing over trivial risk here. Still, that seems to be what you actually searched for, right?)

Comment (slifty) [reply .github/dependabot.yml:29]: I don't remember (I think that's a typo and I checked for both) -- but keep in mind there are eslint packages that definitely *do not* start with eslint (e.g. the typescript-namespaced eslint packages) https://github.com/PhilanthropyDataCommons/service/blob/4e0aacb74207eecdf15c6b7869d379ca41ad17c0/package.json#L54

We'll be using those when we pull out of react app, so I do think there is a practical reason to include it.

**Review by hminsky2002 — APPROVED** [review]
Seems as though this has been worked out!

**Comment by slifty** [comment]
Let the record show that JUSTIN I DO NOT NEED YOUR APPROVAL.

**Pull request merged**